Mathlib/Logic/Equiv/Option.lean

```diff
@@ -0,0 +1,264 @@
+/-
+Copyright (c) 2021 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.Control.EquivFunctor
+import Mathlib.Data.Option.Basic
+import Mathlib.Data.Subtype
+import Mathlib.Logic.Equiv.Defs
+
+/-!
+# Equivalences for `Option α`
+
+
+We define
+* `Equiv.optionCongr`: the `Option α ≃ Option β` constructed from `e : α ≃ β` by sending `none` to
+  `none`, and applying a `e` elsewhere.
+* `Equiv.removeNone`: the `α ≃ β` constructed from `Option α ≃ Option β` by removing `none` from
+  both sides.
+-/
+
+
+namespace Equiv
+
+open Option
+
+variable {α β γ : Type _}
+
+section OptionCongr
+
+/-- A universe-polymorphic version of `EquivFunctor.map_equiv Option e`. -/
+@[simps apply]
+def optionCongr (e : α ≃ β) : Option α ≃ Option β where
+  toFun := Option.map e
+  invFun := Option.map e.symm
+  left_inv x := (Option.map_map _ _ _).trans <| e.symm_comp_self.symm ▸ congr_fun Option.map_id x
+  right_inv x := (Option.map_map _ _ _).trans <| e.self_comp_symm.symm ▸ congr_fun Option.map_id x
+#align equiv.option_congr Equiv.optionCongr
+
+@[simp]
+theorem optionCongr_refl : optionCongr (Equiv.refl α) = Equiv.refl _ :=
+  ext <| congr_fun Option.map_id
+#align equiv.option_congr_refl Equiv.optionCongr_refl
+
+@[simp]
+theorem optionCongr_symm (e : α ≃ β) : (optionCongr e).symm = optionCongr e.symm :=
+  rfl
+#align equiv.option_congr_symm Equiv.optionCongr_symm
+
+@[simp]
+theorem optionCongr_trans (e₁ : α ≃ β) (e₂ : β ≃ γ) :
+    (optionCongr e₁).trans (optionCongr e₂) = optionCongr (e₁.trans e₂) :=
+  ext <| Option.map_map _ _
+#align equiv.option_congr_trans Equiv.optionCongr_trans
+
+/-- When `α` and `β` are in the same universe, this is the same as the result of
+`EquivFunctor.map_equiv`. -/
+theorem optionCongr_eq_equivFunction_mapEquiv {α β : Type _} (e : α ≃ β) :
```

Comment:
Preexisting from mathlib3, but should this be `equivFunctor` instead of `equivFunction`? CC @eric-wieser

Reply:
Yes, please fix the name.